vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/daily_cap_frequency_cap_unittest.cc

```diff
@@ -0,0 +1,193 @@
+ protected:
+  BraveAdsDailyCapFrequencyCapTest() :
+      mock_ads_client_(std::make_unique<MockAdsClient>()),
+      ads_(std::make_unique<AdsImpl>(mock_ads_client_.get())) {
+    // You can do set-up work for each test here
+  }
+
+  ~BraveAdsDailyCapFrequencyCapTest() override {
+    // You can do clean-up work that doesn't throw exceptions here
+  }
+
+  // If the constructor and destructor are not enough for setting up and
+  // cleaning up each test, you can use the following methods
+
+  void SetUp() override {
+    // Code here will be called immediately after the constructor (right before
+    // each test)
+
+    auto callback = std::bind(
+        &BraveAdsDailyCapFrequencyCapTest::OnAdsImplInitialize, this, _1);
+    ads_->Initialize(callback);
+
+    client_mock_ = std::make_unique<ClientMock>(ads_.get(),
+        mock_ads_client_.get());
+    frequency_capping_ = std::make_unique<FrequencyCapping>(client_mock_.get());
+    exclusion_rule_ = std::make_unique<DailyCapFrequencyCap>
+        (frequency_capping_.get());
+  }
+
+  void OnAdsImplInitialize(const Result result) {
+    EXPECT_EQ(Result::SUCCESS, result);
+  }
+
+  void TearDown() override {
+    // Code here will be called immediately after each test (right before the
+    // destructor)
+  }
+
+  std::unique_ptr<MockAdsClient> mock_ads_client_;
+  std::unique_ptr<AdsImpl> ads_;
+
+  std::unique_ptr<ClientMock> client_mock_;
+  std::unique_ptr<FrequencyCapping> frequency_capping_;
+  std::unique_ptr<DailyCapFrequencyCap> exclusion_rule_;
+  AdInfo ad_info_;
+};
+
+TEST_F(BraveAdsDailyCapFrequencyCapTest, AdAllowedWhenNoAds) {
+  // Arrange
+  ad_info_.campaign_id = kTestCampaignIDs.at(0);
+  ad_info_.daily_cap = 2;
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsDailyCapFrequencyCapTest, AdAllowedWithAds) {
+  // Arrange
+  ad_info_.campaign_id = kTestCampaignIDs[0];
+  ad_info_.daily_cap = 2;
+
+  client_mock_->GenerateCampaignHistory(
+      ad_info_.campaign_id, 0, 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsDailyCapFrequencyCapTest, AdAllowedWithAdsWithinTheDay) {
+  // Arrange
+  ad_info_.campaign_id = kTestCampaignIDs[0];
+  ad_info_.daily_cap = 2;
+
+  // 23hrs 59m 59s ago
+  client_mock_->GenerateCampaignHistory(
+      ad_info_.campaign_id, -(kSecondsPerDay - 1), 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsDailyCapFrequencyCapTest, AdAllowedWithAdsOverTheDay) {
+  // Arrange
+  ad_info_.campaign_id = kTestCampaignIDs[0];
+  ad_info_.daily_cap = 2;
+
+  // 24hs ago
+  client_mock_->GenerateCampaignHistory(
+      ad_info_.campaign_id, -kSecondsPerDay, 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
```

Comment:
nit: `const bool`

Reply:
Happy to change, of note elsewhere I've used `auto` - is there a style preference? 

`auto is_ad_excluded = ...`
or 
`const bool is ad_excluded = ...`

Reply:
Personal style, but I am going to move away from `auto` for simple types

Reply:
Adopted `const bool` style